perfkitbenchmarker/providers/azure/azure_virtual_machine.py

```diff
@@ -198,9 +198,10 @@ def _Create(self):
     create_cmd = [AZURE_PATH,
                   'vm',
                   'create',
-                  '--affinity-group=%s' % self.network.affinity_group.name,
+                  '--location=%s' % self.zone,
                   '--virtual-network-name=%s' % self.network.vnet.name,
                   '--vm-size=%s' % self.machine_type,
+                  '--public-ip=%s' % self.name,
```

Comment:
Could you explain how this differs from the current method? Is the named public IP cleaned up when the instance is deleted?